tsinfer/formats.py

```diff
@@ -2225,6 +2215,256 @@ def populations(self):
+
+    def __metadata_schema_getter(self, zarr_group):
+        try:
+            return self.data[zarr_group].attrs["metadata_schema"]
+        except KeyError:
+            return {"codec": "json"}
+
+    @property
+    def format_name(self):
+        return self.FORMAT_NAME
+
+    @property
+    def format_version(self):
+        return self.FORMAT_VERSION
+
+    @property
+    def finalised(self):
+        return True
+
+    @property
+    def sequence_length(self):
+        try:
+            return self.data.attrs["sequence_length"]
+        except KeyError:
+            return int(np.max(self.data["variant_position"])) + 1
+
+    @property
+    def num_sites(self):
+        return self._num_sites
+
+    @property
+    def sites_metadata_schema(self):
+        return self.__metadata_schema_getter("sites")
+
+    @property
+    def sites_metadata(self):
+        try:
+            return self.data["sites/metadata"]
+        except KeyError:
+            return zarr.array(
+                [{}] * self.num_individuals, object_codec=numcodecs.JSON()
+            )
+
+    @property
+    def sites_time(self):
+        try:
+            return self.data["sites/time"]
+        except KeyError:
+            return np.full(self.data["variant_position"].shape, tskit.UNKNOWN_TIME)
+
+    @property
+    def sites_position(self):
+        return self.data["variant_position"]
+
+    @property
+    def sites_alleles(self):
+        return self.data["variant_allele"]
+
+    @property
+    def sites_ancestral_allele(self):
+        try:
+            return self.data["sites/ancestral_allele"]
+        except KeyError:
+            # Maintains backwards compatibility: in previous tsinfer versions the
+            # ancestral allele was always the zeroth element in the alleles list
+            return np.zeros(self.num_sites, dtype=np.int8)
+
+    @property
+    def sites_genotypes(self):
+        gt = self.data["call_genotype"]
+        return gt[...].reshape(gt.shape[0], gt.shape[1] * gt.shape[2])
+
+    @property
+    def provenances_timestamp(self):
+        try:
+            return self.data["provenances_timestamp"]
+        except KeyError:
+            return np.array([], dtype=object)
+
+    @property
+    def provenances_record(self):
+        try:
+            return self.data["provenances_record"]
+        except KeyError:
+            return np.array([], dtype=object)
+
+    @property
+    def num_samples(self):
+        return self._num_samples
+
+    @property
+    def samples_individual(self):
+        ret = np.zeros((self.num_samples), dtype=np.int32)
+        for p in range(self.ploidy):
+            ret[p :: self.ploidy] = np.arange(self.num_individuals)
+        return ret
+
+    @property
+    def metadata_schema(self):
+        try:
+            return self.data.attrs["metadata_schema"]
+        except KeyError:
+            None
+
+    @property
+    def metadata(self):
+        try:
+            return self.data.attrs["metadata_schema"]
```

Comment:
Typo, should be "metadata"

Reply:
https://github.com/tskit-dev/tsinfer/pull/748/commits/e0df71ad5a99ca54f8f17c0920bc76aba8812843